ext/mysqli/mysqli_nonapi.c

```diff
@@ -85,7 +90,9 @@ void mysqli_common_connect(INTERNAL_FUNCTION_PARAMETERS, bool is_real_connect, b
 		}
 
 		if (object) {
+#if ZEND_DEBUG
```

Comment:
This isn't part of the fix but initially I had changed the code in this if branch as well and put this in an `#if` to avoid the overhead, I can drop this or commit this separately

Reply:
I don't think the assert should be in `#if`.

Reply:
> I don't think the assert should be in #if.

I can drop this. The motivation is that the call to instanceof_function_slow will still be done even in release builds, but it's not that big of a deal.